subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/logcat/LogcatAccessor.kt

```diff
@@ -0,0 +1,12 @@
+package com.avito.instrumentation.internal.logcat
+
+internal interface LogcatAccessor {
+
+    val isAvailable: Boolean
+
+    /**
+     * Logcat will be populated with empty lists if not available;
+     * Check [isAvailable] to avoid useless work
```

Comment:
It's easy to forget about another one "if".
It'd be good to express this by types, e.g. sealed class.

Reply:
https://github.com/avito-tech/avito-android/pull/1036#discussion_r643462103